runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -11808,15 +11808,21 @@ static TR::Register *inlineIntrinsicIndexOf(TR::Node *node, bool isLatin1, TR::C
    // to each String.indexOf call is ' ', which is the most common whitespace character.
       {
       TR::Register *value = srm->findOrCreateScratchRegister();
+      TR::Register *zxTargetScalar = srm->findOrCreateScratchRegister();
+
+      // Since we're going to do a load followed immediately by a comparison, we need to ensure that
```

Comment:
So you think we only need to do that for the first character and not for the rest (in the vector loop) ?

Reply:
That's correct. Only the first comparison is done using a 32-bit scalar comparison operator. It doesn't matter what the upper bits of this register are when performing the vector comparison, as they will have been overwritten when the value is splatted into the vector register.